src/main/java/com/aws/greengrass/builtin/services/pubsub/PubSubIPCEventStreamAgent.java

```diff
@@ -76,7 +79,19 @@ public PublishToTopicOperationHandler getPublishToTopicHandler(OperationContinua
      * @param serviceName name of the service subscribing.
      */
     public void subscribe(String topic, Consumer<PublishEvent> cb, String serviceName) {
-        handleSubscribeToTopicRequest(topic, serviceName, cb);
+        subscribe(topic, cb, serviceName, null);
+    }
+
+    /**
+     * Handle the subscription request from internal plugin services with given receive mode.
+     *
+     * @param topic       topic name.
+     * @param cb          callback to be called for each published message
+     * @param serviceName name of the service subscribing.
+     * @param receiveMode the mode of messages sending back to the originating component.
+     */
+    public void subscribe(String topic, Consumer<PublishEvent> cb, String serviceName, ReceiveMode receiveMode) {
```

Comment:
Add a SubscriptionRequest object to encapsulate these options so that if we need to add more again, we don't need to add more subscribe overloads in the future.

```java
SubscriptionRequest {
String topic;
Consumer<PublishEvent> cb;
String serviceName;
ReceiveMode receiveMode;
}
```

Reply:
Added to both subscribe and unsubscribe